library/core/src/ffi/mod.rs

```diff
@@ -92,56 +92,46 @@ pub type c_ssize_t = isize;
 mod c_char_definition {
     cfg_if! {
         // These are the targets on which c_char is unsigned.
-        if #[cfg(any(
-            all(
-                target_os = "linux",
-                any(
-                    target_arch = "aarch64",
-                    target_arch = "arm",
-                    target_arch = "hexagon",
-                    target_arch = "powerpc",
-                    target_arch = "powerpc64",
-                    target_arch = "s390x",
-                    target_arch = "riscv64",
-                    target_arch = "riscv32",
-                    target_arch = "csky"
-                )
+        // See isSignedCharDefault() in clang/lib/Driver/ToolChains/Clang.cpp
```

Comment:
The comment above is no longer correct - at the end of this `if` block, you put `i8`